hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/PlacementPolicy.java

```diff
@@ -22,12 +22,13 @@
 import java.io.IOException;
 import java.util.Collections;
 import java.util.List;
+import java.util.Set;
 
 /**
  * A PlacementPolicy support choosing datanodes to build
  * pipelines or containers with specified constraints.
  */
-public interface PlacementPolicy {
+public interface PlacementPolicy<Replica> {
```

Comment:
I still don't believe we need to make this interface generic. As it stands we are only working with `ContainerReplica` objects, so lets drop the `<Replica>` part here. If we need to make the class generic in the future we can do that, for now, its not needed.

Reply:
This is needed since the placement policy is defined hdds common while ContainerReplica is defined in hdds-scm

Reply:
Don't want Hdds common to have a dependency on hdds-scm. We could use generic for this